src/main/java/binnie/genetics/craftgui/WindowGeneBank.java

```diff
@@ -42,93 +38,98 @@
+		ItemStack converted = Engineering.addGene(held, gene);
+		getPlayer().inventory.setItemStack(converted);
+		getPlayer().inventory.markDirty();
+		if (getPlayer() instanceof EntityPlayerMP) {
+			((EntityPlayerMP) getPlayer()).sendContainerToPlayer(player.inventoryContainer);
+		}
 	}
 
 	@Override
 	public void initialiseServer() {
-		final GeneTracker tracker = GeneTracker.getTracker(this.getWorld(), this.getUsername());
+		GeneTracker tracker = GeneTracker.getTracker(getWorld(), getUsername());
 		if (tracker != null) {
-			tracker.synchToPlayer(this.getPlayer());
+			tracker.synchToPlayer(getPlayer());
 		}
 	}
 
 	@Override
 	public void initialiseClient() {
-		this.setTitle("Gene Bank");
-		this.addEventHandler(new EventValueChanged.Handler() {
+		setTitle("Gene Bank");
+		addEventHandler(new EventValueChanged.Handler() {
 			@Override
-			public void onEvent(final EventValueChanged event) {
+			public void onEvent(EventValueChanged event) {
 				if (event.value instanceof BreedingSystem) {
-					WindowGeneBank.this.genes.setValue((BreedingSystem) event.value);
+					genes.setValue((BreedingSystem) event.value);
 				}
 			}
 		});
-		int boxX = 100;
+
+		int boxX;
 		int x = 16;
-		final int y = 32;
+		int y = 32;
 		new ControlPlayerInventory(this, x, y);
 		x += 124;
 		boxX = x;
-		final int geneBoxWidth = 120;
+		int geneBoxWidth = 120;
 		new Panel(this, boxX + 24, 32.0f, geneBoxWidth, 120.0f, MinecraftGUI.PanelType.Black);
 		new Panel(this, boxX + 24 + geneBoxWidth, 32.0f, 14.0f, 120.0f, MinecraftGUI.PanelType.Gray);
-		final ControlScrollableContent scroll = new ControlScrollableContent(this, boxX + 24 + 2, 34.0f, geneBoxWidth + 10, 116.0f, 12.0f);
-		final ControlTextEdit edit = new ControlTextEdit(this, boxX + 27 + geneBoxWidth - 70, 18.0f, 80.0f, 12.0f);
-		this.addEventHandler(new EventTextEdit.Handler() {
+		ControlScrollableContent scroll = new ControlScrollableContent(this, boxX + 24 + 2, 34.0f, geneBoxWidth + 10, 116.0f, 12.0f);
+		ControlTextEdit edit = new ControlTextEdit(this, boxX + 27 + geneBoxWidth - 70, 18.0f, 80.0f, 12.0f);
+		addEventHandler(new EventTextEdit.Handler() {
 			@Override
-			public void onEvent(final EventTextEdit event) {
-				WindowGeneBank.this.genes.setFilter(event.getValue());
+			public void onEvent(EventTextEdit event) {
+				genes.setFilter(event.getValue());
 			}
 		}.setOrigin(EventHandler.Origin.Self, edit));
-		scroll.setScrollableContent(this.genes = new ControlGeneScroll(scroll, 1.0f, 1.0f, geneBoxWidth, 116.0f));
-		this.genes.setGenes(Binnie.Genetics.beeBreedingSystem);
-		final ControlTabBar<BreedingSystem> tabBar = new ControlTabBar<BreedingSystem>(this, boxX, 32.0f, 24.0f, 120.0f, Position.Left) {
+		scroll.setScrollableContent(genes = new ControlGeneScroll(scroll, 1.0f, 1.0f, geneBoxWidth, 116.0f));
+		genes.setGenes(Binnie.Genetics.beeBreedingSystem);
+		ControlTabBar<BreedingSystem> tabBar = new ControlTabBar<BreedingSystem>(this, boxX, 32.0f, 24.0f, 120.0f, Position.Left) {
```

Comment:
IntelliJ has a useful refactoring tool that helps cut down on the large amount of indentation from the anonymous method here, 
`right click -> Refactor -> Convert Anonymous to Inner`
I've been using it a bunch on the 1.11 branch.